iot_config/src/route_service.rs

```diff
@@ -267,9 +293,17 @@ impl iot_config::Route for RouteService {
                 }
             }
 
-            while let Ok(update) = route_updates.recv().await {
-                if shutdown_listener.is_triggered() || (tx.send(Ok(update)).await).is_err() {
-                    break;
+            tracing::info!("existing routes sent; streaming updates as available");
+            loop {
+                let shutdown = shutdown_listener.clone();
+
+                tokio::select! {
+                    _ = shutdown => break,
+                    msg = route_updates.recv() => if let Ok(update) = msg {
+                        if tx.send(Ok(update)).await.is_err() {
+                            break;
+                        }
+                    }
```

Comment:
pretty sure the previous way i handled streams here i was never going to check if the shutdown listener were triggered until i received a message on the stream and this is why the server would always hang when i tried to ctrl-c it; i believe the new approach with `select!` should resolve that